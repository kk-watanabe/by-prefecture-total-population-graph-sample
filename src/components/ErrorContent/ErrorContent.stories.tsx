import { ErrorContent } from "./ErrorContent";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "components/ErrorContent",
  component: ErrorContent,
} as ComponentMeta<typeof ErrorContent>;

const Template: ComponentStory<typeof ErrorContent> = (args) => <ErrorContent {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: (
    <>
      エラーを表示します。
      <br />
      エラーを表示します。
    </>
  ),
};
