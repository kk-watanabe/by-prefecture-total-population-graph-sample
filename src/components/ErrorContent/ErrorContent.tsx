import styles from "./Styles.module.css";

export interface ErrorContentProps {
  children: JSX.Element;
}

export const ErrorContent = (props: ErrorContentProps) => (
  <div className={styles["error-content"]}>{props.children}</div>
);
