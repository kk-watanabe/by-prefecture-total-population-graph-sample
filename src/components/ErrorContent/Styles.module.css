.error-content {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
  min-height: 200px;
  border: 1px solid var(--color-gray300);
  border-radius: 8px;
  color: var(--color-gray700);
  text-align: center;
  line-height: 1.75;
}
