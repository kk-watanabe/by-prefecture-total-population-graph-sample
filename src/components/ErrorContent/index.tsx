/* istanbul ignore next */
export * from "./ErrorContent";
