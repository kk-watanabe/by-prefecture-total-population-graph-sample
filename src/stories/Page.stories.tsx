import React from "react";

import { Page } from "./Page";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "Example/Page",
  component: Page,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: "fullscreen",
  },
} as ComponentMeta<typeof Page>;

const Template: ComponentStory<typeof Page> = (args) => <Page {...args} />;

export const LoggedOut = Template.bind({});

export const LoggedIn = Template.bind({});
