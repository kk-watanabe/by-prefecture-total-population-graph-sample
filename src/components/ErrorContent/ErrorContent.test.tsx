import { composeStories } from "@storybook/testing-react";
import { render } from "@testing-library/react";

import * as stories from "./ErrorContent.stories";

const { Default } = composeStories(stories);

describe("snapshots", () => {
  it("Readner Default components", () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
