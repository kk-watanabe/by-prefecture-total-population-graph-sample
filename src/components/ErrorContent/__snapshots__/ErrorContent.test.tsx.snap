// Vitest Snapshot v1

exports[`snapshots > Readner Default components 1`] = `
<div
  class="_error-content_rl669_1"
>
  エラーを表示します。
  <br />
  エラーを表示します。
</div>
`;
